test/unit/controller/write-off-controller.ts

```diff
@@ -0,0 +1,202 @@
+/**
+ *  SudoSOS back-end API service.
+ *  Copyright (C) 2024  Study association GEWIS
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as published
+ *  by the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+import { DefaultContext, defaultContext, finishTestDB } from '../../helpers/test-helpers';
+import { truncateAllTables } from '../../setup';
+import WriteOff from '../../../src/entity/transactions/write-off';
+import { seedWriteOffs } from '../../seed';
+import { seedRoles } from '../../seed/rbac';
+import User, { UserType } from '../../../src/entity/user/user';
+import { ADMIN_USER, inUserContext, UserFactory } from '../../helpers/user-factory';
+import { expect, request } from 'chai';
+import { WriteOffResponse } from '../../../src/controller/response/write-off-response';
+import WriteOffController from '../../../src/controller/write-off-controller';
+import TokenMiddleware from '../../../src/middleware/token-middleware';
+import BalanceService from '../../../src/service/balance-service';
+import { json } from 'body-parser';
+import VatGroup from '../../../src/entity/vat-group';
+
+function writeOffEq(a: WriteOff, b: WriteOffResponse): Boolean {
+  return a.to.id === b.to.id
+    && a.amount.getAmount() === b.amount.amount
+    && a.createdAt.toISOString() === b.createdAt
+    && a.updatedAt.toISOString() === b.updatedAt;
+}
+
+describe('WriteOffController', () => {
+  let ctx: DefaultContext & {
+    writeOffs: WriteOff[];
+    adminToken: string;
+    token: string;
+  };
+
+  before(async () => {
+    const c = { ...await defaultContext() };
+    await truncateAllTables(c.connection);
+
+    const admin = await (await UserFactory(await ADMIN_USER())).get();
+    const localUser = await (await UserFactory()).get();
+
+    const all = { all: new Set<string>(['*']) };
+    await seedRoles([{
+      name: 'Admin',
+      permissions: {
+        WriteOff: {
+          create: all,
+          get: all,
+        },
+      },
+      assignmentCheck: async (user: User) => user.type === UserType.LOCAL_ADMIN,
+    }]);
+
+    const adminToken = await c.tokenHandler.signToken({ user: admin, roles: ['Admin'], lesser: false }, 'nonce admin');
```

Comment:
Please use the `await getToken()` helper method, because that creates the token and correctly assigns the roles in the database.